pkg/cert/store.go

```diff
@@ -45,38 +45,56 @@ func NewSecretCertStore(namespace, secretName string, clientset clientset.Interf
 	}
 }
 
+// Will attempt to read a secret from the Kubernetes Secret store, format and return it.
+// Will error out if the secret isn't set, or the certificate inside is not properly formatted.
 func (s *secretCertStore) Current() (*tls.Certificate, error) {
+	// Try to get a Certificate from Kube Secret.
+	// This will be there if there's been a successful deployment of the webhook before in this cluster.
 	secret, err := s.clientset.CoreV1().Secrets(s.namespace).Get(
 		s.secretName,
 		metav1.GetOptions{},
 	)
+	// Create an appropriate custom Error for if the secret doesn't exist.
 	noKeyErr := certificate.NoCertKeyError(
-		fmt.Sprintf("no cert/key files read at secret %s/%s",
+		fmt.Sprintf("Error: No data found for cert/key files at secret %s/%s",
 			s.namespace,
 			s.secretName))
+	// If we had an error getting the secret, exit.
 	if err != nil {
 		klog.Errorf("Error fetching secret: %v", err.Error())
+		klog.Error("This is expected if this is the initial deploy in this cluster.")
```

Comment:
these might not appear after each other in case logs get printed async by various things , I would merge it into one message